ports/esp32/machine_timer.c

```diff
@@ -36,7 +36,9 @@
 #include "modmachine.h"
 
 #define TIMER_INTR_SEL TIMER_INTR_LEVEL
-#define TIMER_DIVIDER  40000
+#define TIMER_DIVIDER  40
```

Comment:
If this was "4" could it support 100ns resolution?

Reply:
A value of 4 would give us 50ns resolution (we currently have 500ns resolution). According to the [ESP32 technical reference manual](https://www.espressif.com/sites/default/files/documentation/esp32_technical_reference_manual_en.pdf) (chapter 18) there is a 16-bit prescaler and then a 64-bit counter and the lowest acceptable value for the prescaler is 2. This means that we could have 25ns resolution if we wanted and still have a range of about 14,600 years, which is probably enough. There might be a slight power consumption penalty for doing this since the prescaler does very little but there is a large stack of comparators and whatnot connected to the main counter, but I doubt that it will be measurable in the grand scheme of things.

Reply:
Ok, thanks for the details.  Probably 40 (500ns resolution) is a good number because the jitter from other parts of the system (eg caching effects) would drown out higher resolution.  Well, it's easy to tune in the future if needed.

Reply:
I tried switching it to 8 (100ns resolution) and I could'd detect any difference in power consumption. Given how unlikely it is for anyone to leave a micro-controller running for 50,000 years, no matter how good Micropython is, I can't see a reason not to leave it that small.

Reply:
I'm happy either way.